templates/multistage-buildconfig.yaml

```diff
@@ -0,0 +1,99 @@
+objects:
+- apiVersion: image.openshift.io/v1
+  kind: ImageStream
+  metadata:
+    name: jlinked-image
+  spec:
+    lookupPolicy:
+      local: false
+- apiVersion: build.openshift.io/v1
+  kind: BuildConfig
+  metadata:
+    name: ubi9-jlinked-image-buildconfig
+  spec:
+    source:
+      dockerfile: |
+        #Stage-1: ubi9-jlinked-image is builder image + application + jlinked JVM
+        # install additional system dependencies (for ubi-micro) to /mnt/jrootfs
+        FROM -
+        USER 0
+        RUN mkdir -p /mnt/jrootfs
+        RUN microdnf install --installroot /mnt/jrootfs --releasever 9 --setopt install_weak_deps=0 --nodocs -y \
+            --config=/etc/dnf/dnf.conf \
+            --noplugins \
+            --setopt=cachedir=/var/cache \
+            --setopt=reposdir=/etc/yum.repos.d \
+            --setopt=varsdir=/etc/dnf/vars \
+            grep gawk
+        RUN rm -rf /mnt/jrootfs/var/cache/* /mnt/jrootfs/var/lib/rpm /mnt/jrootfs/var/lib/dnf
+      strategy:
+        dockerStrategy:
+          from:
+            kind: ImageStreamTag
+            name: quay.io/repository/jdowland/jlink:ubi9-jlinked-image
+      output:
+        to:
+          kind: ImageStreamTag
+          name: ubi9-jlinked-image:latest
+      triggers:
+        - type: ConfigChange
+        - type: ImageChange
+          imageChange:
+            from:
+              kind: ImageStreamTag
+              name: quay.io/repository/jdowland/jlink:ubi9-jlinked-image  # ImageStreamTag to trigger the build (stage-1 in Dockerfile)
+- apiVersion: build.openshift.io/v1
+  kind: BuildConfig
+  metadata:
+    name: lean-runtime-buildconfig
+  spec:
+    source:
+      dockerfile: |
+        #Stage-2:copy application JAR and jlinked JRE to runtime image
+        FROM -
+
+        COPY --from=ubi9-jlinked-image /mnt/jrootfs/ /
+        COPY --from=ubi9-jlinked-image /deployments /deployments
+        COPY --from=ubi9-jlinked-image /tmp/jre ${JAVA_HOME}
+        COPY --from=ubi9-jlinked-image /opt/jboss/container/ /opt/jboss/container/
+
+        # these are in the micro image
+        RUN rm -rf /var/lib/dnf /var/lib/rpm
+
+        USER 185
+        CMD /opt/jboss/container/java/run/run-java.sh
+      strategy:
+        dockerStrategy:
+          from:
+            kind: ImageStreamTag
+            name: registry.access.redhat.com/ubi9/ubi-micro
+        dockerArgs:
+          - name: "JAVA_HOME"
+            value: "/usr/lib/jvm/java"
+        env:
+          - name: "JAVA_HOME"
+            value: "${JAVA_HOME}"
+          - name: "PATH"
+            value: "${JAVA_HOME}/bin:$PATH"
+      output:
+        to:
+          kind: ImageStreamTag
+          name: jlinked-image:latest
+      triggers:
+        - type: ConfigChange
+        - type: ImageChange
+          imageChange:
+            from:
+              kind: ImageStreamTag
+              name: registry.access.redhat.com/ubi9/ubi-micro  # ImageStreamTag for the final output image (stage-2 in Dockerfile)
```

Comment:
this isn't a valid ImageStreamTag. If we wan't to trigger a rebuild when ubi-micro is changed, we will need to define a ubi-micro ImageStream. Perhaps we should